### ISSUE TITLE
Update event histograms for summary 507 - 11 May 2016 fails (hangs)

### ISSUE THREAD

**Issue by tomkooij** [HiSPARC/publicdb#139]
```hisparc-update.py``` currently fails when updating the event histogram for summary 507 - 11 May 2016.

I had to kill the python process that was creating the histogram. It had been running for several days, and prevented new daily updates to run.

After killing the process and resetting the generator state, I tried just running ```hisparc-update``` again, but it failed again. The process in the logfile below seems to hang:
```
2016-05-14 22:27:15,754 - INFO - [17294] - Updating event histograms for Summary: 507 - 11 May 2016
```

I'll copy the esd file to my VM and try to debug. For now, I have unchecked the 'needs update events' flag in summary 507 - 11 May 2016.

**Comment by tomkooij** [comment]
event direction reconstruction "hangs" at:
```
event['id'] == 23240
event['timestamp'] == 1462978820 
```
At that timestamp the events "change" from 2 detectors to 4 detectors. It seems something "happened" to station 507 at 10/11 May.

It'll try to figure out why "bad data" makes it hang, when I have some more time.

**Comment by 153957** [comment]
Possibly due to the strange detector arrangement of 507.. I do not see what you mean by 'from 2 to 4 detectors'. There are earlier events that day on which 4 detectors were hit, for example:
```
2016-05-11	11:21:50	1462965710	473926535	548	1481	2983	2114	9304	21899	152013	35483	8.2033	16.87	97.472	22.297	97.5	65	82.5	77.5	80	-999	-999
```

**Comment by tomkooij** [comment]
My bad: In debugging, I was seeing many reconstructions of events with len(t) == 2 and suddenly (at event_id == 23240) an event with len(t) == 4. And the reconstruction hangs just there. I was under the impression that there were no events with 4 detectors hit before that timestamp.

But indeed, there are (a few) events with 4 detectors hit throughout the day. And a few before the timestamp. Those did not cause an infinite loop.

**Comment by 153957** [comment]
Fixed with HiSPARC/sapphire#131
